### PR TITLE
feat(key-remove): add methods to remove key using pattern

### PR DESCRIPTION
**Pull request by t3ns0r-ss**
Add methods to remove keys using a pattern.
 - TryRemoveAllKeysByPattern method uses library methods to fetch and delete keys one by one
 - TryRemoveAllKeysByPatternUsingLua uses Lua Script to remove the keys.
Pattern should be fed to the methods.

### CODE REVIEW

File: FA.Cache/Providers/RedisCacheProvider.cs [hunk 1]
```diff
@@ -77,5 +80,36 @@ public void TryRemove(string cacheKey)
                 Console.Error.WriteLine(ex);
             }
         }
+
+        /// <summary>
+        /// This method removes keys on the basis of pattern.<br/>
```
Comment (apgapg) [inline FA.Cache/Providers/RedisCacheProvider.cs]: Direct start from /// Removes

File: FA.Cache/Providers/RedisCacheProvider.cs [hunk 2]
```diff
@@ -77,5 +80,36 @@ public void TryRemove(string cacheKey)
...
+            {
+                foreach (var key in _server.Keys(pattern: pattern))
+                {
+                    _cache.KeyDelete(key);
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method removes keys on the basis of pattern using LUA.<br/>
```
Comment (apgapg) [inline FA.Cache/Providers/RedisCacheProvider.cs]: Direct start from /// Removes

File: FA.Cache/Providers/RedisCacheProvider.cs [hunk 3]
```diff
@@ -8,11 +8,14 @@ namespace FA.Cache.Providers
...
     {
         private readonly IDatabase _cache;
+        private readonly IServer _server;
 
         public RedisCacheProvider(string redisConnectionString)
         {
             Console.WriteLine("RedisCacheProvider: Setting up redis connection...");
             var redis = ConnectionMultiplexer.Connect(ConfigurationOptions.Parse(redisConnectionString));
+            //Assuming only one server
+            _server = redis.GetServers().FirstOrDefault();
```
Comment (apgapg) [inline FA.Cache/Providers/RedisCacheProvider.cs:18]: We sure to use this way only? Are we sure this is official way?
any supportive link?

File: FA.Cache/Providers/RedisCacheProvider.cs [hunk 4]
```diff
@@ -8,11 +8,14 @@ namespace FA.Cache.Providers
     public class RedisCacheProvider : ICacheProvider
     {
         private readonly IDatabase _cache;
+        private readonly IServer _server;
 
         public RedisCacheProvider(string redisConnectionString)
         {
             Console.WriteLine("RedisCacheProvider: Setting up redis connection...");
             var redis = ConnectionMultiplexer.Connect(ConfigurationOptions.Parse(redisConnectionString));
+            //Assuming only one server
```
Comment (apgapg) [inline FA.Cache/Providers/RedisCacheProvider.cs]: space after //

**Review by apgapg — CHANGES_REQUESTED** [review]

File: FA.Cache/Providers/RedisCacheProvider.cs (hunk 3 above)
Comment (t3ns0r-ss) [reply FA.Cache/Providers/RedisCacheProvider.cs:18]: https://stackexchange.github.io/StackExchange.Redis/KeysScan#so-how-do-i-use-them

File: FA.Cache/Providers/RedisCacheProvider.cs (hunk 4 above)
Comment (t3ns0r-ss) [reply FA.Cache/Providers/RedisCacheProvider.cs]: Fixed

File: FA.Cache/Providers/RedisCacheProvider.cs (hunk 2 above)
Comment (t3ns0r-ss) [reply FA.Cache/Providers/RedisCacheProvider.cs]: Done.

File: FA.Cache/Providers/RedisCacheProvider.cs (hunk 1 above)
Comment (t3ns0r-ss) [reply FA.Cache/Providers/RedisCacheProvider.cs]: Done.

File: FA.Cache/Providers/RedisCacheProvider.cs (hunk 3 above)
Comment (t3ns0r-ss) [reply FA.Cache/Providers/RedisCacheProvider.cs:18]: Or we can use Lua based modifications

**Review by apgapg — APPROVED** [review]
LGTM

**Pull request merged**